src/actions/deviceActions.js

```diff
@@ -470,10 +470,16 @@ export function setDeviceRunning(isRunning) {
 
 export function setPowerMode(isSmuMode) {
     return async dispatch => {
-        await device.ppkSetPowerMode(isSmuMode ? 2 : 1);
         logger.info(`Mode: ${isSmuMode ? 'Source meter' : 'Ampere meter'}`);
-        dispatch(setPowerModeAction(isSmuMode));
-        if (!isSmuMode) dispatch(setDeviceRunning(true));
+        if (isSmuMode) {
+            dispatch(setDeviceRunning(false));
```

Comment:
I think this should also get an `await`, so that we first finished sending the `DUT off` before we change to source mode.